### ISSUE TITLE
hows this coming along?

### ISSUE THREAD

**Issue by ubuntuaddicted** [tolga9009/elgato-gchd#2]
I really want to help out any way I can, I have arch and xubuntu 14.04.1. Let me know how I can help move this along faster. :)

**Comment by OniLink** [comment]
Any updates? Or just continue doing the same thing as before?

**Comment by tolga9009** [comment]
Hi,

I didn't had time to work on it over the past weeks due to exams. I don't know when I will have enough time to continue my work (probably in about 2 weeks). However, I've bought a dedicated Hackintosh machine now and the v2.0 drivers are working; plus, I have some ideas, what to do. I'm just lacking time at the moment.

If you have any ideas  or pull requests, just tell me.

Cheers,
Tolga

**Comment by tolga9009** [comment]
HOLY MOTHER OF GOD! YES! I finally got it! It's extremely hacky and probably won't work with your setup, but I was finally able to receive audio & video on my native Linux machine! I've stupidly copy & pasted the output of ov2lusb and wrote a receive_data() function, which writes the bulk data to disk. If you open up the file with VLC, you will see the audio & video data. The driver is currently around ~7200. I'm super sure, that it can be done in way less lines; but I didn't had enough time to track down patterns and create for-loops.

Here is the output: https://drive.google.com/file/d/0B29z6-xPIPLEamVZVERITml2OWs/view?usp=sharing

You will notice, that the output is splitted in 2 halves and these halves are mixed. But that's not a major problem for now. Atleast, audio & video are in sync and it runs super smooth at 60fps.

I'm now pushing my working directory to GitHub, so you can work together with me. I will now work on stripping down unnecessary packets and try to figure out more stuff. Stay tuned.

**Comment by jedahan** [comment]
Wow wow that is great news! I am super excited to start testing this :)

**Comment by Hansaplast** [comment]
Awesome! :+1:

**Comment by ubuntuaddicted** [comment]
This is great news! May I suggest you look into the OBS-studio project on github. They have an awesome piece of open source software for capturing and even live streaming your desktop or an external device (cameras and such that use the V4L driver) and now with this awesome news you can make OBS-studio even Better. I have the original elgato so I'll be testing when I get the chance. Thanks for all your hard work. I'm sure it's time more than hard work but its very kind of you none the less

**Comment by ubuntuaddicted** [comment]
I'd be happy to even report on the status during our next podcast, it's called Linux Tech and Gaming. It airs on Wednesday nights at 7pm CST

**Comment by tolga9009** [comment]
I'm making some progress, deciphering the script file (https://goo.gl/uVQhPT). This file seems to be some sort of script file, which is referring to function calls. Things, I've found out so far:

write commands:
- `slsi <wIndex> <data>` -> bRequest = 0xbc, wValue = 0x0000,...
- `sparam <wIndex> <unknown> <unknown> <data>` -> bRequest = 0xbc, wValue = 0x0001,...
- `scmd <data[2]> <data[3]> <data[4] data[5]>` -> bRequest = 0xb8, wValue = 0x0000, wIndex = 0x0000, data[0] = 0x00, data[1] = 0x00,... (system command -> scmd)

read commands:
- `statechange` -> bRequest = 0xbc, wValue = 0x0800, wIndex = 0x2008, wLength = 2; (used for checking device states, after setting device mode with `scmd`)

I've yet to find out, how `i2c write` and `i2c stat` commands are working. These script files are well documented and definitely an extremely great help for reverse engineering.

I think, `scmd` command is needed for "resetting" support (reset Elgato, without replugging). I think, one needs to set the device to "stop" on cleanup, using state change. Like this:
`write_config6(0xb8, 0x0000, 0x0000, 0x00, 0x00, 0x05, 0x00, 0x00, 0x01);`

If you find anything useful, please share.

Cheers,
Tolga

**Comment by tolga9009** [comment]
Okay, very short update. I was finally able to understand sparam function. It works like this:

sparam <wIndex> <bit_shift left> <set_bits> <data>.

**Comment by StianOby** [comment]
Hi! Just wanted to page you here, since we chatted on the OBS forum. I have an Elgato game capture, and some Linux experience (albeit I am no coder). If you need help testing things, I should be able to help out.

My main goal with this driver is to be able to stream (and capture) from my Xbox One. So far it is looking good, apart from the OBS "bug" of no being able to read the FIFO stream properly. Keep up the good work on the driver, and OBS will probably get the fixes soon. I'll try and help pushing the issue so that the devs notice.

One question: how do you record with the driver as of now? Is there an argument that needs to be passed?

PS: 1080p support would be a great thing to prioritize IMHO, at least if that is easier to fix than the reset function. With 1080p support we would have all basic functionality needed for streamers, at the very low cost of replugging the device after each streaming.

**Comment by tolga9009** [comment]
Hi StianOby,

I don't think that it's actually a bug, that OBS can't read from a FIFO pipe. It's probably just not implemented (yet?), because there was no need so far.

Recording is currently not possible without 3rd party software. You will need software like VLC, which can read the stream and capture it to your harddrive. One big advantage is, that you can live monitor your capture. This driver's main purpose is to experiment with the device, understand it and make it easier to write a kernel driver, with V4L2 support (our ultimate goal). Implementing a reset function would make debugging ALOT easier and more comfortable. On the other hand, 1080p support is something I can do in a couple of hours, whereas the reset function would need days and weeks to finish.

I'm having some time this weekend and will try to get 1080p going.

**Comment by StianOby** [comment]
After today's update of OBS I can confirm that it will read the .ts file created by your driver as expected. Here is a proof-of-concept captured just now using the July 9th release of OBS studio: http://www.twitch.tv/stianoby/c/6945787

I found one new issue, though, which I created a separate bug for: https://github.com/tolga9009/elgato-gchd/issues/7

**Comment by Glog78** [comment]
1080p would be awesome for my PS3 capturing. How much latency is there currently ?

**Comment by LRFLEW** [comment]
I found this project while looking to see if anybody had reversed the protocol for the Elgato card. I have two quick questions about the project.

Does this project support the HD60? I don't own the HD60 personally, but I do wonder if the protocol was kept the same. Either way, you should add something to the ReadMe about it so it doesn't keep getting asked.

Also, is there any plans to support a multi-platform release (or at least make the code easy to port)? As a mac user, I'm really annoyed by the official support from Elgato. The software is lagging behind the Windows version significantly and it doesn't support general video output (so I can't use it with OBS, etc.). I'm also annoyed with how it handles interlaced input, but then again, I don't like the solution in the Windows version either. I'd be willing to contribute, especially for mac-specific things like AVFoundation.

**Comment by tolga9009** [comment]
The latency is about 2 - 3 seconds. The Windows drivers are also around 2 seconds of latency. This is limited by hardware, not software.

This driver doesn't support HD60. I assume, that HD60 and GCHD are totally different and we would need to start from scratch. But I may be wrong, didn't had my hands on one. But I'm sure that a working solution for the GCHD would make reverse engineering the HD60 easier.

Eventhough I'm not working towards multi-platform support, it shouldn't be a problem to port this over to Mac OS X or Windows. libusb is multi-platform. That's just something I'm not focusing on. In the long run, a Linux kernel module is planned (for V4L2). That might be the point, where lots of stuff will break in terms of multi-platform.

**Comment by LRFLEW** [comment]
Thanks for the response @tolga9009. 

I did successfully get this project to compile and run on Mac OS X with no changes (I had to install libusb via Brew for anybody trying to replicate it). While I ran into a few issues (namely #7 and an issue getting VLC to recognize the pipe), I was able to use dd to write the stream to a file and play the file back. There were some anomalies near the start of the stream that show as a few seconds of black that might be the cause of the VLC issue. If that doesn't match what you see when you use the driver, I'd be willing to upload the .ts file I get.

I want to get involved with this project. I have a Mac that dual-boots into Windows 10 as well. I don't know how to log USB data (maybe I can find a util for Windows), but I'm willing to invest time into the project once I figure that out. If you have any suggestions on how to log the USB traffic, please let me know.

**Comment by rosimildo** [comment]
I have a device with PID:VID:
Bus 002 Device 004: ID 0fd9:005d Elgato Systems GmbH
but the init function never completes... so, I guess this device has a complete init sequence...

**Comment by rosimildo** [comment]
completing my previous post... I compiled the code on Ubuntu 14.04, added the new PID:VID, but it hang somewhere in the init function due to a different init sequence being my guess....

**Comment by LRFLEW** [comment]
@rosimildo From my brief experience with the driver, it sounds like it's actually blocking on the open. Because of how the FIFO works, you need to open a read on the FIFO object at `/tmp/elgato_gchd.ts`. You can open it in VLC to view it, or you can just use `dd` to get the data read and unblock the open.

**Comment by rosimildo** [comment]
I passed that phase... I have instrumented the driver a bit with "printf's", and it hang after the FW download. If I comment that part out, it goes to the loop... but all streaming data requests times out.

**Comment by rosimildo** [comment]
I have built a setup to make WIRESHARK and USBPCAP captures. I have built a Windows VM, and have all these packages installed, and was able to make a capture. I would like to get directions what is the best approach to make "real captures" that we could use to reserve engineering the protocol.

**Comment by tolga9009** [comment]
I'm capturing USB traffic with an open source hardware analyzer called OpenVizsla. Unfortunately, Mac OS X doesn't offer much in terms of software solutions to capture USB.

PID 005d looks like a new revision or a new region-exclusive device. When did you buy the Elgato GCHD and in which country did you buy it? Nevertheless, I will add this PID, as it doesn't seem to be different on first sight.

You need to replug your Elgato everytime before you start the driver. There is a device shutdown / reset sequence, which I couldn't capture yet. So everytime you stop the driver, it leaves the device in an undefined state.

Don't skip the firmware download, else it won't work. It needs to be loaded everytime.

About the Windows VM + USBPCAP: I've tried this solution before. My Core i5-3570K didn't had enough juice at 4x3.4GHz to run it smoothly, while capturing. Packets get mixed up or get lost. Also, I've been using a Mac OS X base. So I don't think it will help.

**Comment by rosimildo** [comment]
I bought this device in the USA, and about 6-8 months ago... late 2014 or early Jan... I don't remember.
Wow!, I am surprised this setup won't help. Over the weekend, I will try to see if I can push some video to a server.... more later when I get more results. I think your project got some traction, and I'd love to see it going...

**Comment by LRFLEW** [comment]
I'm surprised that an i5-3570K couldn't handle USBPcap. I was able to run it on my laptop with an i7-4960HQ and it seems to have worked just fine. I got an ~8MB capture file, but I'm not sure what it is that I'm looking at with it. I'm not sure if "packets get mixed up or get lost" because I'm not sure what it's supposed to look like. I can read it with Wireshark (both on Mac OS X and Windows), but it's just a bunch of numbers and hex data, and I don't know what to do with it. If you know of any resources to learn more about how to interpret the USB packets, please let me know. Otherwise, I'll do what I can to get started.

**Comment by tolga9009** [comment]
@rosimildo I've done a bit more digging and was not able to find the 0x005d in my INF files. Could you identify and upload your INF file, so I can compare them? Here is my (latest Elgato GCHD drivers): https://gist.github.com/tolga9009/361d9dfc1824a01e336e.

As you can see, only the 0x0044 and 0x004e are listed. I'm wondering, if they are using the same driver. I've also found out, that the Elgato GCHD is based on the Yuan UB658. They provide an SDK, which is not free.

@LRFLEW I'm not sure why, but performance breaks down, once recording is active. The resulting MPEG file is not clean - there are artefacts and missing frames. Not easy to describe, but it's nowhere near the smooth output of the native Windows drivers. So, I think, that the processor can't handle that much load, eventhough we might overcome this issue with a better VM / faster processor.

Learning how to interpret USB packets was an interactive process to me. I've come across something I didn't know, looked it up and proceeded. But I think http://www.usbmadesimple.co.uk/ is a great website to start with.

To understand the "bunch of numbers and hex data", you need to investigate and understand the Windows / Mac OS X drivers. This is were reverse engineering starts. For example, I was able to identify the firmware upload process and reverse engineer the dlfirm() function by comparing the firmware file, I had found in the Mac drivers with the actual capture log. But there are lots of things, which I couldn't identify. I'm currently working on adapting the cryptic `write_config()` to a more readable format. Hopefully, this will be a point, where things will start to make sense.

**Comment by rosimildo** [comment]
Here it is a lsusb -v for my device:

Bus 001 Device 005: ID 0fd9:005d Elgato Systems GmbH
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0        64
  idVendor           0x0fd9 Elgato Systems GmbH
  idProduct          0x005d
  bcdDevice            2.00
  iManufacturer           1 Elgato
  iProduct                2 GameCapture HD
  iSerial                 3 070D747780
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           81
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              200mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           9
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0
      bInterfaceProtocol      0
      iInterface              0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x85  EP 5 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x86  EP 6 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x07  EP 7 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x88  EP 8 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x89  EP 9 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
Device Qualifier (for other device speed):
  bLength                10
  bDescriptorType         6
  bcdUSB               2.00
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0        64
  bNumConfigurations      1
Device Status:     0x0000
  (Bus Powered)

**Comment by rosimildo** [comment]
Binaries files:

08/18/2014  10:34 AM               256 EDID.bin
09/29/2014  04:34 PM               256 EDID_1080p.bin
01/12/2015  12:59 PM            29,052 Firmware_EGCHD.bin
08/18/2014  10:34 AM            32,240 Firmware_EGCHD2.bin

**Comment by rosimildo** [comment]
Here it is the .INF file:

;==============================================================================
; Elgato Game Capture
; Driver Install Script (32 bit)
; Copyright (c) 2010, Elgato Systems GmbH
;------------------------------------------------------------------------------        
; History:
;------------------------------------------------------------------------------
; 06-Dec-11 MS 1st Elgato version
; 30-Aug-12 MS add support for new HW PID 0x004E
; 12-Nov-12 MS add support for new HW PID 0x0051
; 13-Nov-14 MS add support for new HW PID 0x005D
;------------------------------------------------------------------------------
; Notes:
;------------------------------------------------------------------------------
;
;==============================================================================

[Version]
Signature   = "$CHICAGO$"
Class       = Media
ClassGUID   = {4d36e96c-e325-11ce-bfc1-08002be10318} 
Provider    = %Provider% 
DriverVer	= 11/13/2014,2.3.3.38
CatalogFile = Game_Capture.cat  

[SourceDisksNames]
1 = %DiskName%,,

[SourceDisksFiles]
ElgatoGC658.sys = 1

[DestinationDirs]
EyeTV.CopySys   = 12                  ; driver files

[ControlFlags]
ExcludeFromSelect    = *
ExcludeFromSelect.NT = *

[Manufacturer]
%Provider% = Elgato, NTx86

[Elgato.NTx86]
%EyeTV.DeviceDesc% = EyeTV,USB\VID_0FD9&PID_0044      
%EyeTV.DeviceDesc% = EyeTV,USB\VID_0FD9&PID_004E  
%EyeTV.DeviceDesc% = EyeTV,USB\VID_0FD9&PID_0051 
%EyeTV.DeviceDesc% = EyeTV,USB\VID_0FD9&PID_005D          

[EyeTV.NTx86]
Include       = ks.inf, kscaptur.inf
Needs         = KS.Registration.NT, KSCAPTUR.Registration.NT
AddReg        = EyeTV.AddReg
CopyFiles     = EyeTV.CopySys

[EyeTV.CopySys]
ElgatoGC658.sys

;===============================================================
;                R e g i s t r y
;===============================================================

[EyeTV.AddReg]
HKR,,DevLoader,,*ntkern
HKR,,NTMPDriver,,ElgatoGC658.sys
HKR,,PageOutWhenUnopened,3,01

;===============================================================
;                I n t e r f a c e s
;===============================================================


;===============================================================
;                S e r v i c e s
;===============================================================

[EyeTV.NTx86.Services]
Addservice = ElgatoGC658Y, 0x00000002, EyeTV.ServiceInstall

[EyeTV.ServiceInstall]
DisplayName    = %Service.FriendlyName%
ServiceType    = 1                  ; SERVICE_KERNEL_DRIVER
StartType      = 3                  ; SERVICE_DEMAND_START
ErrorControl   = 1                  ; SERVICE_ERROR_NORMAL
ServiceBinary  = %10%\System32\Drivers\ElgatoGC658.sys
LoadOrderGroup = Base
AddReg         = EyeTV.Service.AddReg

[EyeTV.Service.AddReg]
HKR,,DigitalJmp,0x00010001,0x00

;===============================================================
;                S t r i n g s
;===============================================================

[Strings]
;---------------------------------------------------------------
; Non-localizable (Windows)
;---------------------------------------------------------------
; Registry types
REG_SZ                           = 0x00000000
REG_MULTI_SZ                     = 0x00010000
REG_EXPAND_SZ                    = 0x00020000
REG_DWORD                        = 0x00010001

; Service: ServiceType values
SERVICE_KERNEL_DRIVER            = 0x00000001
SERVICE_FILE_SYSTEM_DRIVER       = 0x00000002
SERVICE_ADAPTER                  = 0x00000004
SERVICE_RECOGNIZER_DRIVER        = 0x00000008
SERVICE_WIN32_OWN_PROCESS        = 0x00000010
SERVICE_WIN32_SHARE_PROCESS      = 0x00000020
SERVICE_INTERACTIVE_PROCESS      = 0x00000100
SERVICE_INTERACTIVE_SHARE_PROCESS= 0x00000120

; Service: StartType values
SERVICE_BOOT_START               = 0x00000000
SERVICE_SYSTEM_START             = 0x00000001
SERVICE_AUTO_START               = 0x00000002
SERVICE_DEMAND_START             = 0x00000003
SERVICE_DISABLED                 = 0x00000004

; Service: ErrorControl values
SERVICE_ERROR_IGNORE             = 0x00000000
SERVICE_ERROR_NORMAL             = 0x00000001
SERVICE_ERROR_SEVERE             = 0x00000002
SERVICE_ERROR_CRITICAL           = 0x00000003

; Characteristic flags
NCF_VIRTUAL                 = 0x0001
NCF_WRAPPER                 = 0x0002
NCF_PHYSICAL                = 0x0004
NCF_HIDDEN                  = 0x0008
NCF_NO_SERVICE              = 0x0010
NCF_NOT_USER_REMOVABLE      = 0x0020
NCF_HAS_UI                  = 0x0080
NCF_MODEM                   = 0x0100

; Service install flags
SPSVCINST_TAGTOFRONT        = 0x1
SPSVCINST_ASSOCSERVICE      = 0x2

;---------------------------------------------------------------
; Localizable
;---------------------------------------------------------------

Provider                 = "Elgato "
DiskName                 = "EyeTV Driver CD"

; Service strings
Service.FriendlyName     = "Elgato Game Capture"

; Device Manager
EyeTV.DeviceDesc         = "Game Capture HD"

**Comment by tolga9009** [comment]
Looks good. Seems to work exactly the same way. It should just work after adding PID.

**Comment by LRFLEW** [comment]
Thanks @tolga9009 for the link, but I don't know how much it's actually helped. I think at this point, I'm more confused with Wireshark than anything. There seems to be a lot of "heartbeat" messages recorded (numerous small similar packets sent often), and despite USBPcap saying that the device was on its own hub, a lot of different "devices" were recorded. I also have no clue whether the hex data at the bottom of the window include the packet header or not. I really do want to help, but I feel like I'm hitting a brick wall trying to figure this out. I know commenting here with questions isn't going to work in the long run, but I don't know where else to go to figure this stuff out.

I will say here that this project as-is is already very helpful. Once #7 is fixed, I can easily use it with OBS for live streaming much easier and better than using the default software (at least on Mac OS X). Thank you for starting and continuing to work on this project :D

**Comment by StianOby** [comment]
Just wanted to leave a small report her from a very successful weekend livestreaming using this elgato-gchd driver on Ubuntu 15.10 w/OBS multiplatform. 

I recently bough a new LCD screen with a separate HDMI out port, which provided a workaround for bug #7. Thus, by using the HDMI out port from the screen and setting my Xbox One to 720p I was able to spectate and livestream a Halo 5 gaming tournament the past weekend.

The driver and OBS ran for hours on end, with no crashes or instability. The number of dropped frames was less than 0,1%. In total I streamed something like 6-7 hours including breaks. Most of the action has been cut out using Twitch's highlight function, and is available on YouTube:
- https://www.youtube.com/watch?v=rJA3NjLjAjo
- https://www.youtube.com/watch?v=wErWYdoW1iE
- https://www.youtube.com/watch?v=O9azUVEMMq4

*In other words*: This driver works very well even at this alpha stage. If 1080p support is added I have basically all I need. For those lacking monitors with an HDMI output (or a lag-free HDMI splitter), however, issue #7 is still of critical importance.

**Comment by tolga9009** [comment]
@StianOby: I've watched the videos on YouTube. It's amazing to see this project beeing actually used in real-world scenarios. Thanks for sharing!

@All: thanks for reporting and all the feedback! I've moved the discussion to https://gitter.im/tolga9009/elgato-gchd and closing this "issue" now.

I finally have a running development environment again and everything is set up for a long reverse engineering session. You can check out a picture of it here: https://twitter.com/CevelNet/status/681284139266519040. Please note, that a third monitor is also set up in the background (for reverse engineering HDMI pass-through) and a Xbox360 is sitting on one of my PCs. It's extremely messy, cables everywhere.

I will keep you updated on Gitter.

Cheers,
Tolga
